.github/workflows/main.yml

```diff
@@ -145,8 +145,21 @@ jobs:
         run: |
           python scripts/make_docs.py
           git diff --exit-code
+      - name: Upload the example installers as artifacts
+        if: github.event_name == 'pull_request' && matrix.python-version == '3.9'
+        uses: actions/upload-artifact@v3
+        with:
+          name: installers-${{ runner.os }}-${{ github.sha }}-${{ github.run_id }}-${{ github.run_number }}-${{ github.run_attempt }}
+          path: "${{ runner.temp }}/examples_artifacts"
+          retention-days: 7
+
+  report:
+    name: Report failures
+    needs: [tests]
+    if: always() && github.event_name == 'push' && needs.tests.result == 'failure'
```

Comment:
Should we also check if we are pushing to `main`, or is that guaranteed? Better to add it even if the `on` triggers are configured safely now, because we might miss it in a future updates and end up with issues for noisy signals.